src/OpenTelemetry.Exporter.Prometheus/PrometheusExporter.cs

```diff
@@ -36,28 +36,20 @@ public class PrometheusExporter : MetricExporter
         public PrometheusExporter(PrometheusExporterOptions options)
         {
             this.Options = options;
-            this.Metrics = new List<Metric>();
+            this.Metrics = Enumerable.Empty<Metric>();
         }
 
-        private List<Metric> Metrics { get; set; }
+        internal IEnumerable<Metric> Metrics { get; private set; }
 
         /// <inheritdoc/>
         public override Task<ExportResult> ExportAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken)
         {
             // Prometheus uses a pull model, not a push.
-            // Accumulate the exported metrics internally, return success.
+            // Update the reference to the current metrics and return success.
             // The pull process will read this internally stored metrics
             // at its own schedule.
-            this.Metrics.AddRange(metrics);
+            this.Metrics = metrics;
```

Comment:
The exporter was accumulating records, because our controller is a 'Push' one - it can occur more frequently than Prometheus scrap. i.e if push occurs every 15 secs, and promtheus scrapes every 60 sec, we want to keep/accumulate the data from 4 pushes, and give it to prometheus.
This caused the multiple instance of same metric, as there was no "merging" in this exporter.

The actual fix would be to use PrometheusPullController, which calls Collect() in response to prometheus scrape call.

Reply:
If the same time-series is present multiple times this will generate data that will result in undefined behavior for Prometheus this is not fixed by adding different timestamps. Typically for Prometheus you just want the latest data. This will be a problem if the metric values are reset between pushes.

Reply:
Ok. This wont be a problem once Pull based model is implemented.